MSSQL Integration - Update documentation with extra windows authentication details (#6325)
grafana/agent: docs/sources/flow/reference/components/prometheus.exporter.mssql.md

See [sql_exporter](https://github.com/burningalchemist/sql_exporter#collectors) for details on how to create a configuration.

### Authentication

> By default, the `USERNAME` and `PASSWORD` used within the `connection_string` should correspond with a SQL Server username and password.

Reviewer: Should or does. Should implies it doesn't necessarily need to. If that's what is being explined here, then it's OK as-is. If it's a _must_, then we need to make sure it's clear.

```suggestion
By default, the `USERNAME` and `PASSWORD` used within the `connection_string` corresponds to an SQL Server username and password.
```

> If the Grafana Agent is running in the same Windows domain as the SQL Server, then the parameter `authenticator=winsspi` can be used within the `connection_string` without any additional credentials in order to authenticate.

Reviewer: ```suggestion
If {{< param "PRODUCT_ROOT_NAME" >}} is running in the same Windows domain as the SQL Server, then you can use the parameter `authenticator=winsspi` within the `connection_string` without any additional credentials to authenticate.
```

```conn
sqlserver://@<HOST>:<PORT>?authenticator=winsspi
```

> Otherwise if you would like to use Windows credentials to authenticate (instead of SQL Server credentials), you can use the parameter `authenticator=ntlm` within the `connection_string`. The `USERNAME` and `PASSWORD` would then be the Windows username and password (domain may need to be prefixed to the username).

Reviewer: ```suggestion
If you want to use Windows credentials to authenticate, instead of SQL Server credentials, you can use the parameter `authenticator=ntlm` within the `connection_string`.
The `USERNAME` and `PASSWORD` would be the Windows username and password.
The Windows domain may need to be prefixed to the username.
```

Doesn't it always need to include the domain?

Author: @clayton-cornell at least in one of the tests that I did, I did not need to add the domain to connect to SQL Server running on a Windows machine from my local Macbook (using admin Windows credentials).

Reviewer: ok, cool. I was just making sure that the `may` part was correct/
